src/App.tsx

```diff
@@ -1,27 +1,79 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { get5First, getAll, getRedGoods } from './api/goods';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const isEmpty = goods.length === 0;
+  const areRed = goods.every((good) => good.color === 'red');
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleLoadAllGoods = async () => {
+    const loadedGoods = await getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+    return (isEmpty || goods.length === 5)
+      ? (
+        setGoods(loadedGoods)
+      )
+      : (
+        setGoods([])
+      );
```

Comment:
i'm confused why would you need all these
just set your state to loadedgoods and that's it
```suggestion
    setGoods(loadedGoods);
```

Reply:
Ah I was just playing that they would work together the bottons and also like if I click again they would hide the list! etc, I will remove all the extra and commit again!